### **E-commerce Analysis with Markdown**
This is a simple data exploration exercise using just markdown. 

Imagine you have a dataset containing information about online store orders. Let's explore some trends!

**Table:**

| order_id  | customer_id | order_date  | order_value ($)   | product_category |
|-----------|-------------|-------------|-------------------|------------------|
| 001       | 23          | 02/04/2024  | 200               | Food             |
| 002       | 25          | 02/06/2024  | 300               | Electronics      |
| 003       | 23          | 02/10/2024  | 1000              | Healthcare       |
| 004       | 18          | 02/18/2024  | 800               | Clothing         |
| 005       | 21          | 03/01/2024  | 3000              | Food             |
| 006       | 05          | 03/03/2024  | 500               | Healthcare       |
| 007       | 29          | 03/04/2024  | 600               | Food             |
| 008       | 10          | 03/08/2024  | 5000              | Electronics      |
| 009       | 22          | 03/12/2024  | 400               | Healthcare       |
| 010       | 05          | 03/15/2024  | 1200              | Clothing         |
| 011       | 19          | 03/18/2024  | 500               | Food             |
| 012       | 05          | 03/26/2024  | 600               | Healthcare       |

**Columns:**

- **order_id**: Unique identifier for the order (text).
- **customer_id**: Unique identifier for the customer (text).
- **order_date**: Date of the order (text, format MM/DD/YYYY).
- **order_value**: Total value of the order (number).
- **product_category**: Category of the main product purchased (text).

**Tasks:**

1. Describe the Data:

- How many orders are there in total?
- What are some summary statistics for the order_value (e.g., mean, median, minimum, maximum)?

2. Explore Trends:

- Are there any differences in average order value by product_category?

**How to Solve:**

- While you cannot perform calculations directly in markdown, you can use it to structure your analysis.

1. Descriptive Statistics:

    - Use markdown headings and text to explain what you're analyzing (e.g., "Total Orders").
    - Get the results of your calculations and write them down (e.g., "Total Orders: 1000").

2. Category Comparison:

    - Create a table with headers "Product Category" and "Average Order Value".
    - Fill the table with the average values of each category (e.g., "Electronics: $ 250").

Remember:
This is a simplified exercise. Real data analysis would involve data cleaning, visualization, and further exploration.

# Descriptive Statistics
## Orders Information


##### Sum of Order Value For Each Product Category
| Product Category| Sum|
| --------------------------------|-|
| Clothing | 2000 |
| Electronics | 5300 |
| Food | 4300 |
| Healthcare | 2500 |


##### Count of Order Per Product Category
| Product Category| Count|
| --------------------------------|-|
| Clothing | 2 |
| Electronics | 2 |
| Food | 4 |
| Healthcare | 4 |

##### Sum of all Order Values
| Sum of all values |
|-|
| 14100 |

##### Least Worth of Order Value Sum

| Category | Sum |
|- | - |
| Clothing | 2000 |

##### Highest Worth of Order Value Sum

| Category | Sum |
|- | - |
| Electronics | 5300 |


##### Average order values 
| Average of all order values |
|-|
| 1175 |

##### Median order values 
| Median of all order values |
|-|
| 600 |

##### Information about the minimum order value
|order_id	|customer_id	|order_date|	order_value|	product_category|	MIN(order_value)|
|-|-|-|-|-|-|
|001|	23	|02/04/2024	|200	|Food	|200|

##### Information about the maximum order value
|order_id	|customer_id	|order_date|	order_value|	product_category|	MAX(order_value)|
|-|-|-|-|-|-|
|008|	10	|03/08/2024	|5000	|Electronics	|5000|


In [1]:
%load_ext sql

In [2]:
%sql sqlite:///exercise_1.db

In [3]:
%%sql
CREATE TABLE IF NOT EXISTS commerce(
    order_id TEXT NOT NULL,
    customer_id TEXT NOT NULL,
    order_date TEXT NOT NULL,
    order_value NUMBER NOT NULL,
    product_category TEXT NOT NULL
);

 * sqlite:///exercise_1.db
Done.


[]

In [4]:
%%sql
INSERT INTO commerce (order_id, customer_id, order_date, order_value, product_category)
VALUES 
    ('001','23','02/04/2024',200, 'Food'),
    ('002','25','02/06/2024',300,'Electronics'),
    ('003','23','02/10/2024',1000,'Healthcare'),
    ('004','18','02/18/2024',800,'Clothing'),
    ('005','21','03/01/2024',3000,'Food'),
    ('006','05','03/03/2024',500,'Healthcare'),
    ('007','29','03/04/2024',600,'Food'),
    ('008','10','03/08/2024',5000,'Electronics'),
    ('009','22','03/12/2024',400,'Healthcare'),
    ('010','05','03/15/2024',1200,'Clothing'),
    ('011','19','03/18/2024',500,'Food'),
    ('012','05','03/26/2024',600, 'Healthcare');

 * sqlite:///exercise_1.db
12 rows affected.


[]

In [5]:
%%sql
SELECT (SUM(order_value))
FROM commerce

 * sqlite:///exercise_1.db
Done.


[(14100,)]

In [18]:
%%sql 
SELECT product_category, SUM(order_value) AS order_value_per_product
FROM commerce
GROUP BY product_category
ORDER BY order_value_per_product DESC
LIMIT 1

 * sqlite:///exercise_1.db
Done.


[('Electronics', 5300)]

In [19]:
%%sql 
SELECT product_category, SUM(order_value) AS order_value_per_product
FROM commerce
GROUP BY product_category
ORDER BY order_value_per_product ASC
LIMIT 1

 * sqlite:///exercise_1.db
Done.


[('Clothing', 2000)]

In [7]:
%%sql
SELECT  product_category, (SUM(order_value)) as order_value_per_product
FROM commerce
GROUP BY product_category

 * sqlite:///exercise_1.db
Done.


[('Clothing', 2000),
 ('Electronics', 5300),
 ('Food', 4300),
 ('Healthcare', 2500)]

In [8]:
%%sql
SELECT product_category, COUNT(product_category) AS count_of_orders
FROM commerce
GROUP BY product_category

 * sqlite:///exercise_1.db
Done.


[('Clothing', 2), ('Electronics', 2), ('Food', 4), ('Healthcare', 4)]

In [9]:
%%sql
SELECT AVG(order_value)
FROM commerce

 * sqlite:///exercise_1.db
Done.


[(1175.0,)]

In [10]:
%%sql 
WITH ranked_order_value AS (
    SELECT order_value, 
           ROW_NUMBER() OVER (ORDER BY order_value ASC) AS row_asc,
           COUNT (*) OVER () AS total_count
    FROM commerce
)
SELECT 
    CASE WHEN total_count / 2.0 LIKE '%.5' THEN (SELECT AVG(order_value) FROM ranked_order_value WHERE row_asc >= FLOOR(total_count /2.0) AND row_asc < FLOOR((total_count / 2.0) + 2)) 
        ELSE (SELECT order_value FROM ranked_order_value WHERE row_asc = (total_count / 2)) 
    END AS Median
FROM ranked_order_value
LIMIT 1



 * sqlite:///exercise_1.db
Done.


[(600,)]

In [11]:
%%sql 
SELECT *
FROM commerce
ORDER BY order_value ASC

 * sqlite:///exercise_1.db
Done.


[('001', '23', '02/04/2024', 200, 'Food'),
 ('002', '25', '02/06/2024', 300, 'Electronics'),
 ('009', '22', '03/12/2024', 400, 'Healthcare'),
 ('006', '05', '03/03/2024', 500, 'Healthcare'),
 ('011', '19', '03/18/2024', 500, 'Food'),
 ('007', '29', '03/04/2024', 600, 'Food'),
 ('012', '05', '03/26/2024', 600, 'Healthcare'),
 ('004', '18', '02/18/2024', 800, 'Clothing'),
 ('003', '23', '02/10/2024', 1000, 'Healthcare'),
 ('010', '05', '03/15/2024', 1200, 'Clothing'),
 ('005', '21', '03/01/2024', 3000, 'Food'),
 ('008', '10', '03/08/2024', 5000, 'Electronics')]

In [12]:
%%sql 
SELECT *, MIN(order_value)
FROM commerce

 * sqlite:///exercise_1.db
Done.


[('001', '23', '02/04/2024', 200, 'Food', 200)]

In [13]:
%%sql 
SELECT *, MAX(order_value)
FROM commerce

 * sqlite:///exercise_1.db
Done.


[('008', '10', '03/08/2024', 5000, 'Electronics', 5000)]

# Category Comparison:
| Product Category | Average Order Value |
| - | - |
|Clothing	|$ 1000.0|
|Electronics	|$ 2650.0|
|Food	|$ 1075.0|
|Healthcare	|$ 625.0|


In [14]:
%%sql
SELECT product_category as 'Product Category', "$ " || printf(AVG(order_value)) as 'Average Order Value'
FROM commerce
GROUP BY product_category

 * sqlite:///exercise_1.db
Done.


[('Clothing', '$ 1000.0'),
 ('Electronics', '$ 2650.0'),
 ('Food', '$ 1075.0'),
 ('Healthcare', '$ 625.0')]


## **SALES REPORT - COMPANY XYZ**

### Order Results
We have here the data of the online orders from the store this past few weeks, shown on Table 1. 

We have a total of **$ 14,100** Order Value for the recorded sales. 

### **Table 1:**

   | order_id  | customer_id | order_date  | order_value ($) | product_category |
   |-----------|-------------|-------------|-------------------|------------------|
   | 001       | 23          | 02/04/2024  | 200               | Food             |
   | 002       | 25          | 02/06/2024  | 300               | Electronics      |
   | 003       | 23          | 02/10/2024  | 1000              | Healthcare       |
   | 004       | 18          | 02/18/2024  | 800               | Clothing         |
   | 005       | 21          | 03/01/2024  | 3000              | Food             |
   | 006       | 05          | 03/03/2024  | 500               | Healthcare       |
   | 007       | 29          | 03/04/2024  | 600               | Food             |
   | 008       | 10          | 03/08/2024  | 5000              | Electronics      |
   | 009       | 22          | 03/12/2024  | 400               | Healthcare       |
   | 010       | 05          | 03/15/2024  | 1200              | Clothing         |
   | 011       | 19          | 03/18/2024  | 500               | Food             |
   | 012       | 05          | 03/26/2024  | 600               | Healthcare       |

#### Table 2 shows the _Product Category_  and the _Average_ customer has purchased to us:

### **Table 2:**

    
   | product_category | order_value ($) | Average order_value ($)  | 
   |------------------|-------------------|----------------------------|
   | Food             | 4300              | 1075                       | 
   | Electronics      | 5300              | 2650                       | 
   | Healthcare       | 2500              |  625                       | 
   | Clothing         | 2000              | 1000                       | 
   |    Total         |    14000          |                            | 

#### In Table 3 we arrived at this Statistical Data, based on the table above:

### **Table 3:**

    
   | Statistical Order_value | Value ($)   | product_category  | 
   |-------------------------|-------------|-------------------|
   | Mean                    | 1175        |                   | 
   | Median                  | 600         |                   | 
   | Minimum                 | 200         |  Food             | 
   | Maximum                 | 5000        | Electronics       | 
   
## (Hypothetical Insights)
1. Customers prefer buying Electronics than other items. This may be due to natural phenomena such as weather and convenience.
2. Since food, clothing, and healthcare are basic necessities for our customers, we must ensure an adequate supply of these items.


# Exercise: Formatting Data into Markdown Tables
Objective:
You will be given a dataset in CSV format. Your task is to convert this dataset into a Markdown table format.

Instructions:

**1. Convert the Dataset:** Convert the provided dataset into a Markdown table format.

**2. Use Markdown Syntax:** Make sure to use proper Markdown table syntax to format the table.

**3. Ensure Readability:** Ensure that the table is properly aligned and readable.

__ID,Name,Age,Department,Salary__

- 1,John Doe,28,Marketing,50000

- 2,Jane Smith,34,Sales,55000

- 3,Emily Davis,29,Engineering,60000

- 4,Michael Brown,42,HR,52000

- 5,Alice Johnson,31,Finance,58000

- 6,Robert White,37,IT,62000

- 7,Jessica Lee,26,Marketing,51000

- 8,David Wilson,45,Sales,57000

- 9,Linda Clark,38,Engineering,61000

- 10,James Lewis,33,Finance,59000

|ID|Name|Age|Department|Salary|
|-|-|-|-|-|
|1|John Doe|28|Marketing|50000|
|2|Jane Smith|34|Sales|55000|
|3|Emily Davis|29|Engineering|60000|
|4|Michael Brown|42|HR|52000|
|5|Alice Johnson|31|Finance|58000|
|6|Robert White|37|IT|62000|
|7|Jessica Lee|26|Marketing|51000|
|8|David Wilson|45|Sales|57000|
|9|Linda Clark|38|Engineering|61000|
|10|James Lewis|33|Finance|59000|